### DISCUSSION TITLE
Setting up Disqus and OpenGraph

### DISCUSSION THREAD

**Discussion by rcdailey** [adityatelange/hugo-PaperMod#508] [Q&A]
I want to check 2 aspects of my configuration for correctness. I'm asking here as opposed to the general Hugo forum because I want to make sure there's nothing theme-specific I need to be considerate of.

First, for my Disqus setup, here is what I have done. Note that in general, anything affecting my blog I try to keep as a theme so it stays modularized and I'm not modifying core folders. I'll admit I'm confused on the difference between themes, templates, and modules, but I assume themes are a fine solution here (I see the disqus comments working after all).

I created a single file under `themes/disqus/layouts/partials/comments.html` with the contents:

```html
{{ template "_internal/disqus.html" . }}
```

In my `config.yml` I add it in (along with others) like this:

```yml
theme:
  - favicon
  - disqus
  - opengraph
  - PaperMod
```

Also in my config.yml:

```yml
disqusShortname: myshortname
params:
  comments: true
```

I have done no additional disqus configuration. The disqus help says there's more to set up to make sure links from email notifications and such work, but I am not sure if I need to do that. Does the built in internal template for Disqus do this? What more do I need to set up?

Regarding OpenGraph: I don't really know what it is. I assume it enables links to my site to have a nice "card" preview like you'd see if you pasted it in Discord or something. But I also got the impression that it affects SEO somehow. So I enabled it by using another theme at `themes/opengraph/layouts/partials/head/custom.html`:

```html
{{ template "_internal/opengraph.html" . }}
```

Is this correct?

I'm really uncertain if there's anything I've just configured wrong on my site, but I guess I'll find out as I go.  Thanks for any help.

**Comment by danielfdickinson** [comment]
hugo-Papermod has OpenGraph support, so no need for custom stuff.

I'll try and update the Wiki with the docs (I see that it's undocumented at present) later tonight.

**Reply by rcdailey** [reply]
The twitter cards as well? So all of that stuff happens out of the box, no need to do anything on my part? How about my Disqus config?

**Reply by danielfdickinson** [reply]
Some config is required, hence my wanting to contribute a PR to the wiki. Yes, Twitter cards are available as well.

I know nothing about Disqus as I haven't used it, so can't comment other than I'm sure I've seen it in the layouts.

**Comment by danielfdickinson** [comment]
I'ved added a PR with documentation https://github.com/adityatelange/hugo-PaperMod/pull/513

**Reply by danielfdickinson** [reply]
FYI #513 was merged, so if you check the doc site you should have the Open-Graph / Twitter information need.  The main thing to keep in mind with Open-Graph is that if you do not have an image many consumers of Open-Graph will 'complain' and not the use the Open-Graph metadata. If you define an Open-Graph image for the website as a whole, you should be able to avoid that because it would be used as a fallback.

HTH.

**Comment by 1mfaasj** [comment]
I also want to add Disqus to my papermod hugo website, but I can't figure it out without doing trial and error. the documentation from hugo and on this repository is not sufficient enough.

apparently I need to add a line in the config.yml, is that correct?

what i have now
`theme: PaperMod`

how to change it:
```
theme:
   - PaperMod
   - disqus
```
and I need to make a change in the comments.html , correct?

is that in:
`...\themes\PaperMod\layouts\partials\...?`
or
`...\blog\layouts\...?`

it would be nice if you could add this in the wiki or documentation on the papermod website, a nice addition to any blogs/websites

**Reply by adityatelange** [reply]
Please don't blame people at Hugo as well as Papermod for not having enough documentation.There is enough documentation. 
It's more like you are lazy to go through all of that.

**Reply by rcdailey** [reply]
I won't over-generalize and say there's not enough documentation like this person did, but I do feel like there is either something missing and/or incorrect, or I lack fundamental understanding required to set up Hugo but is not directly part of it.

For example, the [PaperMod wiki](https://github.com/adityatelange/hugo-PaperMod/wiki/Features#comments) says to paste the Disqus HTML into a `comments.html`, but then links to [the official docs on comments](https://gohugo.io/content-management/comments/), which gives me the impression the HTML is not needed and is all working behind the shortcode.

What I can repeat, which was originally stated in my question here, is that Disqus seems to be working fine on my blog so far (with limited testing). I only followed the Hugo instructions, not the Papermod instructions. I do see a mixture of potentially conflicting information between Papermod & Hugo comment setup instructions, but I don't know if that's a documentation problem or me just not understanding something correctly (likely the latter).

The main reason for my question here was to make sure I'm setting it up correctly.

**Reply by adityatelange** [reply]
> I won't over-generalize and say there's not enough documentation like this person did, but I do feel like there is either something missing and/or incorrect, or I lack fundamental understanding required to set up Hugo but is not directly part of it.
> 
> For example, the [PaperMod wiki](https://github.com/adityatelange/hugo-PaperMod/wiki/Features#comments) says to paste the Disqus HTML into a `comments.html`, but then links to [the official docs on comments](https://gohugo.io/content-management/comments/), which gives me the impression the HTML is not needed and is all working behind the shortcode.
> 
> What I can repeat, which was originally stated in my question here, is that Disqus seems to be working fine on my blog so far (with limited testing). I only followed the Hugo instructions, not the Papermod instructions. I do see a mixture of potentially conflicting information between Papermod & Hugo comment setup instructions, but I don't know if that's a documentation problem or me just not understanding something correctly (likely the latter).
> 
> The main reason for my question here was to make sure I'm setting it up correctly.

I would like to correct you herr Papermod wiki is generalized for all type of comment providers. It is not specific to disqus.

See we don't control hugo documentation from here so it's upto the user to correlate the instructions. Please don't expect each and every thing to be mentioned in the wiki. There is a learning curve for Hugo. It's not a no-code solution.

Lastly I'd say we have a better wiki than most of the Hugo themes out there. I hope you agree.

**Reply by rcdailey** [reply]
Oh I agree you have fantastic documentation. I really have no complaints there. If anything, consider this discussion part of the Hugo learning curve. I recognize Hugo has a steep learning curve and the documentation has been nothing but helpful. Where I get a little confused is how the PaperMod documentation interacts with the Hugo documentation. It's not immediately clear to me. Is it supplementary? Mutually exclusive? I just don't know for sure. Where I really disagree with you is the "Up to the user to correlate the instructions". I think that part needs to be documented to some degree. For example, state that the comments instructions are supplemental to the Hugo docs and generalized for multiple providers. Then it's a little more clear to me that when I use something in vanilla Hugo, I'm not bypassing some theme-specific functionality.

You're a maintainer in this repo so there's a lot of knowledge I'm sure you take for granted. I unfortunately have never even used Go before, so many of these fundamentals are new to me. Thank you for helping!

**Reply by Muhammad-Mohid-Khan** [reply]
> > I won't over-generalize and say there's not enough documentation like this person did, but I do feel like there is either something missing and/or incorrect, or I lack fundamental understanding required to set up Hugo but is not directly part of it.
> > For example, the [PaperMod wiki](https://github.com/adityatelange/hugo-PaperMod/wiki/Features#comments) says to paste the Disqus HTML into a `comments.html`, but then links to [the official docs on comments](https://gohugo.io/content-management/comments/), which gives me the impression the HTML is not needed and is all working behind the shortcode.
> > What I can repeat, which was originally stated in my question here, is that Disqus seems to be working fine on my blog so far (with limited testing). I only followed the Hugo instructions, not the Papermod instructions. I do see a mixture of potentially conflicting information between Papermod & Hugo comment setup instructions, but I don't know if that's a documentation problem or me just not understanding something correctly (likely the latter).
> > The main reason for my question here was to make sure I'm setting it up correctly.
> 
> I would like to correct you herr Papermod wiki is generalized for all type of comment providers. It is not specific to disqus.
> 
> See we don't control hugo documentation from here so it's upto the user to correlate the instructions. Please don't expect each and every thing to be mentioned in the wiki. There is a learning curve for Hugo. It's not a no-code solution.
> 
> Lastly I'd say we have a better wiki than most of the Hugo themes out there. I hope you agree.

theres absolutely no need for this and the OP is right theres no clear explanation on how to configure disqus properly in your documentation just following the Hugo is properly enough according to the guy that replied to you. You dont need to have a big ego just because you wrote a good documentation and then aren't willing to help people out for their genuine questions everything can be improved and nothing is perfect. The purpose of a wiki is to let people know about the particular thing they are using and a github discussions page is for asking questions if they are confused its really not that deep. Either you take constructive criticism for the wiki or you help people out in the discussions.
